service/raft/raft_group_registry.cc

```diff
@@ -190,6 +190,11 @@ void raft_group_registry::init_rpc_verbs() {
 
         return container().invoke_on(shard_for_group(gid),
                 [addr = netw::messaging_service::get_source(cinfo).addr, from, gid, handler] (raft_group_registry& self) mutable {
+            future<> f_wait_for_server = self._group0_id && self._group0_id == gid
+                    ? self.wait_for_group0()
```

Comment:
I cannot say I like the approach. What about not marking the group0 as alive on this node until it is actually created?

Reply:
Could you expand on what you mean by "not marking the group0 as alive on this node"? Is there a way to control liveness on a per-group basis, not a per-node basis?

Or do you mean e.g. modifying the `direct_fd_pinger` so that it considers the node as down until it starts its group0 server?

Reply:
> Could you expand on what you mean by "not marking the group0 as alive on this node"? Is there a way to control liveness on a per-group basis, not a per-node basis?
> 
> Or do you mean e.g. modifying the `direct_fd_pinger` so that it considers the node as down until it starts its group0 server?

Yes. That what I mean.

Reply:
I see. Now I also see that we have an issue about it: https://github.com/scylladb/scylladb/issues/15167. I'll try and see whether I'll be able to fix two issues with one PR.